// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

message MessageFromServer {
  int32 id = 1;
  string msg = 2;
}

message MessageFromClient {
  int32 id = 1;
  string msg = 2;
}

// The greeting service definition.
service HelloService  {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc ClientStreamMessages (stream MessageFromClient) returns (MessageFromServer) {}
  rpc ServerStreamMessages (MessageFromClient) returns (stream MessageFromServer) {}
  rpc BidirectionalStreamMessages (stream MessageFromClient) returns (stream MessageFromServer) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}
